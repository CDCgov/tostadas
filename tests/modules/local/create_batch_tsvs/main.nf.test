nextflow_process {

    name "Test Process CREATE_BATCH_TSVS"
    script "modules/local/create_batch_tsvs/main.nf"
    process "CREATE_BATCH_TSVS"

    test("Should split Excel into TSV batches") {

        when {
            params {
                // no parameters needed
            }

            process {
                """
                input[0] = file("${projectDir}/tests/modules/local/create_batch_tsvs/test_genbank.xlsx")
                input[1] = 1
                """
            }
        }

        then {
            assert process.success
            assert !process.out.tsv_files.isEmpty()
            assert snapshot(process.out.tsv_files[0]).match()
        }
    }
}
