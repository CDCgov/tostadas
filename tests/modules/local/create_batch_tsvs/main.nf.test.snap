{
    "Should split Excel into TSV batches": {
        "content": [
            "genbank_batch_1.tsv:md5,e2475926c845dc27cb4609c1923b7766"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-19T00:57:19.369820824"
    }
}